suite-native/atoms/src/Input/SearchInput.tsx

```diff
@@ -68,24 +66,30 @@ export const SearchInput = ({
 }: InputProps) => {
     const { applyStyle, utils } = useNativeStyles();
     const [isFocused, setIsFocused] = useState<boolean>(false);
+    const [isClearButtonVisible, setIsClearButtonVisible] = useState<boolean>(false);
     const searchInputRef = useRef<TextInput | null>(null);
-    const isClearButtonVisible = !!value.length;
     const handleClear = () => {
+        setIsClearButtonVisible(false);
+        searchInputRef.current?.clear();
         onChange('');
     };
 
     const handleInputFocus = () => {
         searchInputRef?.current?.focus();
     };
 
+    const handleOnChangeText = (value: string) => {
+        setIsClearButtonVisible(!!value.length);
+        onChange(value);
+    };
+
     return (
         <Pressable onPress={handleInputFocus}>
             <Box style={applyStyle(inputWrapperStyle, { isFocused, elevation })}>
                 <Icon name="search" color="iconSubdued" size="large" />
                 <TextInput
                     ref={searchInputRef}
-                    value={value}
```

Comment:
Why did you changed this to uncontrolled input?

Reply:
This was causing the text not to render. 

And previous implementation only make the value go round: it took the value, sent it to parent through `onChange()`, parent saved it, used length to decide if to show/hide the clear button and sent it back to the TextInput.

Reply:
I've examined this yesterday and played a lot with it, but have not found a different solution.

Reply:
That's very strange, it's basically standart controlled input which should work no matter what... but let's don't burn more time on it